lib/gnat/services/service_responder.ex

```diff
@@ -0,0 +1,317 @@
+defmodule Gnat.Services.ServiceResponder do
```

Comment:
In this project overall I've been writing GenServers with a structure of:

1. Public Interface Functions
2. GenServer Callbacks
3. Private functions

I think Dave Thomas was the first person I heard this from, but I've seen it around in the community. Can we organize this module in a similar way?